### ISSUE TITLE
Check the need to retry again

### ISSUE THREAD

**Issue by dms120** [normartin/ts-retry-promise#35]
It would be nice to check the error to know if it's worth retrying or not.
Some option like **shouldRetry** or **retryIf**:

```
delay: 1000,
shouldRetry: (error) => {
   return (error.httpCode === 429);
}
```

It's a cool feature that a lot of us would find useful.
Best regards.

**Comment by normartin** [comment]
@dms120 sorry for the delay. How does this #36 look like?

**Comment by dms120** [comment]
Hi @normartin looks perfect! Thank you

**Comment by normartin** [comment]
@dms120 welcome :)

released version [0.7.0](https://www.npmjs.com/package/ts-retry-promise/v/0.7.0)
